src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,220 @@
+                    if (e.next == null) {
+                        newTab[e.hash & (newCap - 1)] = e;
+                    } else {
+                        MyHashMap.Node<K,V> loHead = null;
+                        MyHashMap.Node<K,V> loTail = null;
+                        MyHashMap.Node<K,V> hiHead = null;
+                        MyHashMap.Node<K,V> hiTail = null;
+                        MyHashMap.Node<K,V> next;
+                        do {
+                            next = e.next;
+                            if ((e.hash & oldCap) == 0) {
+                                if (loTail == null) {
+                                    loHead = e;
+                                } else {
+                                    loTail.next = e;
+                                }
+                                loTail = e;
+                            } else {
+                                if (hiTail == null) {
+                                    hiHead = e;
+                                } else {
+                                    hiTail.next = e;
+                                }
+                                hiTail = e;
+                            }
+                        } while ((e = next) != null);
+
+                        if (loTail != null) {
+                            loTail.next = null;
+                            newTab[j] = loHead;
+                        }
+                        if (hiTail != null) {
+                            hiTail.next = null;
+                            newTab[j + oldCap] = hiHead;
+                        }
+                    }
+                }
+            }
+        }
+        return newTab;
+    }
+
+    MyHashMap.Node<K,V> newNode(int hash, K key, V value, MyHashMap.Node<K,V> next) {
+        return new MyHashMap.Node<>(hash, key, value, next);
     }
 
     @Override
     public V getValue(K key) {
+        Node<K,V> e;
+        return (e = getNode(key)) == null ? null : e.value;
+    }
+
+    final MyHashMap.Node<K,V> getNode(Object key) {
+        MyHashMap.Node<K,V>[] tab;
+        MyHashMap.Node<K,V> first;
+        MyHashMap.Node<K,V> e;
+        int n;
+        int hash;
+        K k;
+
+        if ((tab = table) != null && (n = tab.length) > 0
+                && (first = tab[(n - 1) & (hash = hash(key))]) != null) {
+            if (first.hash == hash
+                    && ((k = first.key) == key || (key != null && key.equals(k)))) {
+                return first;
+            }
+            if ((e = first.next) != null) {
+                do {
+                    if (e.hash == hash
+                            && ((k = e.key) == key || (key != null && key.equals(k)))) {
+                        return e;
+                    }
+                } while ((e = e.next) != null);
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class Node<K, V> {
```

Comment:
The Node class is an inner class, and you have access to its private fields directly. Therefore, using getters and setters is redundant. According to the checklist, don't use getters and setters in the Node class.